### PR TITLE
removing husky - fixes #152

### LINKED ISSUE #152: Postinstall prevents installation as a dependency
The package.json specifies a postinstall which initializes husky.  Upon an npm install this postinstall step triggers husky which attempts to locate a '.git' folder within the node module.  This step fails as it is not a git repo resulting in a failed module installation.
See: https://github.com/typicode/husky/issues/884

### PR DESCRIPTION
**Pull request by TheMcMurder**
Fixes #152

I find the new Husky packages hard to use so I'm switching this project to use `lefthook` because it's a lot easier to manage IMHO instead of updating husky to the latest recommendations.

### CODE REVIEW

File: lefthook.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+pre-commit:
+  commands:
+    prettier:
+      glob: "*.{js,jsx}"
+      run: prettier --write {staged_files} && git add {staged_files}
+    test:
+      run: pnpm run test
```
Comment (TheMcMurder) [inline lefthook.yml:7]: One really nice thing about `lefthook` is that you don't need `pretty-quick` dependency anymore which removes a ton of additional dependencies without losing any functionality.

**Comment by joeldenning** [comment]
I prefer keeping all the esm-bundle repos using the same dependencies, and updating the template repo whenever any changes are made. I don't have issues with husky or pretty-quick, could you clarify why you're wanting to change to them?

**Comment by TheMcMurder** [comment]
The specific issue that brought me to this is that trying to do a `yarn 1` install of `@esm-bundle/react-dom` fails.

`yarn add --dev @esm-bundle/react-dom --latest --exact`

```
error [file-system-path]: Command failed.
Exit code: 127
Command: husky install
Arguments: 
Directory: [file-system-path]
Output:
/bin/sh: husky: command not found
```

This appears to be related to `husky install` being called when the package is installed. The current recommendation for fixing this within `husky` is to[ use `prepare` instead of `postinstall`](https://github.com/typicode/husky/issues/884) but that doesn't work in yarn 2 at all.

# Choices

## `prepare` instead of `postinstall`
Pros:
- one line fix

Cons:
- Doesn't work in `yarn v2`

## Lefthook

Pros:
- Works in `yarn v1`, `npm`, and `yarn v2`
- Fewer required dependencies

Cons:
- Multiple line fix
- different dependencies

# Why I chose lefthook

- Opinion: Husky has had a lot of breaking changes in the last year or so and I've found it becoming increasingly difficult to use (pinst stuff as an example)
- Opinion: I've been using lefthook at work at love it
- Performance: I've noticed at work that lefthook is much faster than husky
- Opinion: I find lefthook generally easier to reason about
- Maintenance: Removing a bunch of dependencies seems like a good idea because it will make renovate less likely to encounter issues upgrading

**Comment by TheMcMurder** [comment]
Just adding that if the primary concern is consistency I'd be willing to spend time over the next few weeks migrating repos over to `lefthook` and updating the template as well.

**Comment by TheMcMurder** [comment]
Turns out this isn't needed to fix #152, we just needed to publish a `.fix` version.

**Pull request closed without merging**